sdk/nodejs/provider/server.ts

```diff
@@ -401,6 +426,46 @@ class Server implements grpc.UntypedServiceImplementation {
     }
 }
 
+function configureRuntime(req: any, engineAddr: string) {
+    // NOTE: these are globals! We should ensure that all settings are identical between calls, and eventually
+    // refactor so we can avoid the global state.
+    runtime.resetOptions(req.getProject(), req.getStack(), req.getParallel(), engineAddr,
+                                req.getMonitorendpoint(), req.getDryrun());
+
+    const pulumiConfig: {[key: string]: string} = {};
+    const rpcConfig = req.getConfigMap();
+    if (rpcConfig) {
+        for (const [k, v] of rpcConfig.entries()) {
+            pulumiConfig[k] = v;
+        }
+    }
+    runtime.setAllConfig(pulumiConfig);
```

Comment:
Will need to add configSecretKeys here https://github.com/pulumi/pulumi/pull/7358

Reply:
I included this when I rebased against master.